Source/SwiftyDropbox/Shared/Handwritten/OAuth/OAuthUtils.swift

```diff
@@ -25,4 +25,15 @@ enum OAuthUtils {
         ])
         return params
     }
+
+    // Extracts query parameters from URL and removes percent encoding.
+    static func extractParamsFromUrl(_ url: URL) -> [String: String] {
```

Comment:
We should make the return type something more manageable for the rest of the code base. `extractParamsFromUrl` should return some structure that we can use throughout this code base to do checks in a more Object Oriented way to simplify the usage. Otherwise, we end up with all this string checking into this `results` dictionary, which by the way, the naming is really confusing. For example, I would return a structure like `OauthParameters` :

```
enum OauthState {
    case oauth2 // "oauth2:\(nonce)"
    .... etc..
}

enum OauthError {
    case accessDenied 
    case canceled
    .... etc..
}

struct OauthParameters {
    let oauthState: OauthState 
    let error: OauthError?
    init(parametersMap: [String:String]) {
        /// ... init ivars
    }
}
```

Reply:
great point, will clean this up. thanks!